### PR TITLE
Update notes and links to dev container spec

### PR DESCRIPTION
**Pull request by bamurtaugh**
Updates to how and when we link to the spec. Updated a few links to point to https://containers.dev/ rather than https://github.com/devcontainers/spec as I think containers.dev might be easier to start with and is a centralized set of everything related to the spec.

### CODE REVIEW

**Review by chrmarti — DISMISSED** [review]
Sounds good, thanks!

**Comment by bamurtaugh** [comment]
Thanks @chrmarti!

@gregvanl this is ready for your review and merge whenever you're ready.

File: docs/devcontainers/containers.md [hunk 1]
```diff
@@ -22,9 +22,11 @@ The Dev Containers extension supports two primary operating models:
 * You can use a container as your full-time development environment
 * You can [attach to a running container](/docs/devcontainers/attach-container.md) to inspect it.
 
+> **Note**: The Dev Containers extension supports the open Dev Containers Specification, which empowers anyone in any tool to configure a consistent dev environment. You can learn more in our [dev container FAQ](./faq.md#can-i-use-dev-containers-outside-of-vs-code) and on the specification's site [containers.dev](https://containers.dev/).
```
Comment (gregvanl) [inline docs/devcontainers/containers.md:25]: Can you up this to (/docs/devcontainers/faq.md#can-i-use-dev-containers-outside-of-vs-code)

**Comment by gregvanl** [comment]
@bamurtaugh Looks good. One relative doc link to fix then I can approve and merge

File: docs/devcontainers/containers.md (hunk 1 above)
Comment (bamurtaugh) [reply docs/devcontainers/containers.md]: Thanks! Just updated.

**Review by gregvanl — APPROVED** [review]
Thanks!

**Pull request merged**